File: javascript/leetcode/replace-words.js
/**
 * @param {string[]} dict
 * @param {string} sentence
 * @return {string}
 */
var replaceWords = function(dict, sentence) {
    shortened = sentence.split(' ')
    for (var i = 0; i < shortened.length; i++) {
        for (var j = 0; j < dict.length; j++) {
            if (shortened[i].indexOf(dict[j]) == 0) {
                shortened[i] = dict[j]
            }
        }
    }
    return shortened.join(' ')
};

File: python/leetcode/replace-words.py
class Solution:
    def replaceWords(self, dict, sentence):
        """
        :type dict: List[str]
        :type sentence: str
        :rtype: str
        """
        shortened = sentence.split(' ')
        for i in range(len(shortened)):
            for d in dict:
                try:
                    if shortened[i].index(d) == 0:
                        shortened[i] = d
                except:
                    continue
        return ' '.join(shortened)
